File: OsmAnd/src/net/osmand/plus/track/GradientScaleType.java
package net.osmand.plus.track;

import android.content.Context;

import androidx.annotation.DrawableRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;

import net.osmand.plus.R;
import net.osmand.shared.routing.RouteColorize.ColorizationType;

public enum GradientScaleType {

	SPEED("speed", "gradient_speed_color", R.string.shared_string_speed, R.drawable.ic_action_speed),
	ALTITUDE("altitude", "gradient_altitude_color", R.string.altitude, R.drawable.ic_action_altitude_average),
	SLOPE("slope", "gradient_slope_color", R.string.shared_string_slope, R.drawable.ic_action_altitude_ascent);

	private final String typeName;
	private final String colorTypeName;
	@StringRes
	private final int resId;
	@DrawableRes
	private final int iconId;

	GradientScaleType(@NonNull String typeName, @NonNull String colorTypeName, @StringRes int resId, @DrawableRes int iconId) {
		this.typeName = typeName;
		this.colorTypeName = colorTypeName;
		this.resId = resId;
		this.iconId = iconId;
	}

	public String getTypeName() {
		return typeName;
	}

	public String getColorTypeName() {
		return colorTypeName;
	}

	public int getIconId() {
		return iconId;
	}

	public String getHumanString(@NonNull Context ctx) {
		return ctx.getString(resId);
	}

	public ColorizationType toColorizationType() {
		if (this == SPEED) {
			return ColorizationType.SPEED;
		} else if (this == ALTITUDE) {
			return ColorizationType.ELEVATION;
		} else if (this == SLOPE) {
			return ColorizationType.SLOPE;
		} else {
			return ColorizationType.NONE;
		}
	}

	@Nullable
	public static GradientScaleType getGradientTypeByName(@Nullable String name) {
		for (GradientScaleType scaleType : values()) {
			if (scaleType.name().equalsIgnoreCase(name)) {
				return scaleType;
			}
		}
		return null;
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/gpx/GradientScaleType.kt
package net.osmand.shared.gpx

import net.osmand.shared.routing.RouteColorize
import net.osmand.shared.util.Localization

enum class GradientScaleType(
	val typeName: String,
	val colorTypeName: String,
	val resId: String,
	val iconId: String
) {
	SPEED(
		"speed",
		"gradient_speed_color",
		"shared_string_speed",
		"ic_action_speed"
	),
	ALTITUDE(
		"altitude",
		"gradient_altitude_color",
		"altitude",
		"ic_action_altitude_average"
	),
	SLOPE(
		"slope",
		"gradient_slope_color",
		"shared_string_slope",
		"ic_action_altitude_ascent"
	);

	fun getHumanString(): String = Localization.getString(resId)

	fun toColorizationType(): RouteColorize.ColorizationType {
		return when (this) {
			SPEED -> RouteColorize.ColorizationType.SPEED
			ALTITUDE -> RouteColorize.ColorizationType.ELEVATION
			SLOPE -> RouteColorize.ColorizationType.SLOPE
			else -> RouteColorize.ColorizationType.NONE
		}
	}

	companion object {
		fun getGradientTypeByName(name: String?): GradientScaleType? {
			return entries.find { it.name.equals(name, ignoreCase = true) }
		}
	}
}
